azkaban-common/src/main/java/azkaban/version/JdbcVersionSetLoaderImpl.java

```diff
@@ -0,0 +1,217 @@
+            }
+
+            // Query for versionSet again after insert
+            versionSets = transOperator.query(SELECT_VSET_FROM_MD5, versionSetHandler, versionSetMd5Hex);
+            if (versionSets.size() > 1) {
+                throw new SQLException("Expected only one VersionSet for newly inserted versionSetMd5Hex " + versionSetMd5Hex);
+            }
+            if (versionSets.isEmpty()) {
+                logger.info("No VersionSet exists for newly inserted versionSetMd5Hex " + versionSetMd5Hex);
+                return null;
+            }
+            return versionSets.get(0).getVersionSetNum();
+
+        };
+        try {
+            Integer versionSetNum = this.dbOperator.transaction(insertVersionSet);
+            VersionSet versionSet = new VersionSet(versionSetJsonString, versionSetMd5Hex, versionSetNum);
+            this.md5ToVersionSet.put(versionSetMd5Hex, versionSet);
+            this.numToVersionSet.put(versionSetNum, versionSet);
+            return versionSetNum;
+        } catch (SQLException e) {
+            logger.error("Exception occurred while inserting version set and getting version number", e);
+            throw new IOException("Unable to insert and get versionSetNum", e);
+        }
+    }
+
+    /**
+     * Removes the versionSet row from the table version_set corresponding to versionSetMd5Hex.
+     */
+    @Override
+    public synchronized boolean removeVersionSet(String versionSetMd5Hex) throws IOException {
+        final SQLTransaction<Integer> insertVersionSet = transOperator -> transOperator.update(DELETE_VSET, versionSetMd5Hex);
+        try {
+            Integer rowsDeleted = this.dbOperator.transaction(insertVersionSet);
+            if (this.md5ToVersionSet.containsKey(versionSetMd5Hex)) {
+                VersionSet versionSet = this.md5ToVersionSet.get(versionSetMd5Hex);
+                this.md5ToVersionSet.remove(versionSetMd5Hex);
+                this.numToVersionSet.remove(versionSet.getVersionSetNum());
+            }
+            return rowsDeleted != 0;
+        } catch (SQLException e) {
+            logger.error("Failed to execute: " + DELETE_VSET, e);
+            throw new IOException("Exception occurred while removing version set for " + versionSetMd5Hex, e);
+        }
+    }
+
+    /**
+     * This method first checks if the versionSetNum already exists in the local copy of the versionSets,
+     * otherwise, it insert the versionSet into the table and get the resulting versionSetNum.
+     */
+    @Override
+    public synchronized int getVersionSetNum(String versionSetJsonString, String versionSetMd5Hex) throws IOException {
+        if (this.md5ToVersionSet.containsKey(versionSetMd5Hex)) {
+            VersionSet versionSet = this.md5ToVersionSet.get(versionSetMd5Hex);
+            return versionSet.getVersionSetNum();
+        }
+        return insertAndGetVersionSetNum(versionSetJsonString, versionSetMd5Hex);
```

Comment:
Nit: Can you please change the order of parameters? (versionSetMd5Hex, versionSetJsonString)

Reply:
Done